testnet/stacks-node/src/tests/nakamoto_integrations.rs

```diff
@@ -976,6 +992,34 @@ fn correct_burn_outs() {
         .stop_chains_coordinator();
     run_loop_stopper.store(false, Ordering::SeqCst);
 
+    let stacker_sets = test_observer::get_stacker_sets();
+    info!("Stacker sets announced {:#?}", stacker_sets);
+    let mut sorted_stacker_sets = stacker_sets.clone();
+    sorted_stacker_sets.sort_by_key(|(_block_id, cycle_num, _reward_set)| *cycle_num);
+    assert_eq!(
+        sorted_stacker_sets, stacker_sets,
+        "Stacker set should be sorted by cycle number already"
+    );
+
+    let first_epoch_3_cycle = burnchain
+        .block_height_to_reward_cycle(epoch_3.start_height)
+        .unwrap();
+    for (_, cycle_number, reward_set) in stacker_sets.iter() {
```

Comment:
Any way to expand this to e.g. 10 stackers, some of which don't make the stacking minimum?

Reply:
Yes, but I think that kind of thing is better handled by a unit test in the `make_signer_set` code. Integration tests are very cumbersome for that (not just to write, but to run and maintain) because it will definitely flap if its making very tight assertions (and it should). This is exactly the unit tests that I was waiting to add until I'd gotten some initial feedback on the approach.